### PR TITLE
src: fixes linking source redirection

### PR DESCRIPTION
**Pull request by chris-ramon**
- Improvements on `Dashboard` component, to redirect after sources are resolved.
- Make sure getSources action is promise based.

### CODE REVIEW

File: src/actions/source.ts [hunk 1]
```diff
@@ -65,30 +66,33 @@ export function deleteSource(source: Source): (dispatch: Redux.Dispatch<any>) =>
...
+                    for (const k of keys) {
+                        promises.push(Bluebird.resolve(service.getSource(k)));
+                    }
+                }
+                Bluebird.all(promises.map((p) => p.reflect()))
+                    .each((inspection: Bluebird.Inspection<any>) => {
+                        if (inspection.isFulfilled()) {
+                            sources.push(new Source(inspection.value().val()));
+                        } else {
+                            console.log(inspection.reason());
```
Comment (AnkRaiza) [inline src/actions/source.ts]: are we gonna leave this log here?

File: src/frames/Dashboard.tsx [hunk 2]
```diff
@@ -99,24 +99,22 @@ class Dashboard extends React.Component<DashboardProps, DashboardState> {
...
         if (!pipe.diagnosticsKey) {
           await SpokeService.savePipe(self.props.user, source, pipe.http, true);
         }
-        redirect();
-      } catch (err) {
-        redirect();
-      }
-    } else {
-      this.props.goTo("/skills");
+      } catch (err) {}
```
Comment (AnkRaiza) [inline src/frames/Dashboard.tsx]: isn't this generating an unused variable error?

File: src/index.tsx [hunk 3]
```diff
@@ -95,14 +96,31 @@ Firebase.auth().onAuthStateChanged(function (user: Firebase.User) {
  *
  * See below on the onEnter method.
  */
-let checkAuth: EnterHook = function (nextState: RouterState, replace: RedirectFunction) {
+let checkAuth = function (nextState: RouterState, replace: RedirectFunction): boolean {
     const session: any = store.getState().session;
     if (!session.user) {
```
Comment (AnkRaiza) [inline src/index.tsx]: maybe just
!session.user && replace({...});
return !!session.user;

File: src/index.tsx [hunk 4]
```diff
@@ -95,14 +96,31 @@ Firebase.auth().onAuthStateChanged(function (user: Firebase.User) {
...
             query: nextState.location.query,
             state: { nextPathName: nextState.location.pathname, query: nextState.location.query }
         });
+        return false;
+    }
+    return true;
+};
+
+let onEnterDashboard: EnterHook = function (nextState: RouterState, replace: RedirectFunction) {
+    if (!checkAuth(nextState, replace)) {
```
Comment (AnkRaiza) [inline src/index.tsx]: no need for brackets here if(!chekauth()) return;

**Comment by AnkRaiza** [comment]
minor comments

File: src/actions/source.ts (hunk 1 above)
Comment (chris-ramon) [inline src/actions/source.ts]: yes, we need to log, when something went wrong while monitoring a source.

File: src/frames/Dashboard.tsx (hunk 2 above)
Comment (chris-ramon) [inline src/frames/Dashboard.tsx]: it is a unused variable, but we should keep it to complain with tslint

File: src/index.tsx (hunk 3 above)
Comment (chris-ramon) [inline src/index.tsx]: Interesting suggestion, but I think it reads better as it is

File: src/index.tsx (hunk 4 above)
Comment (chris-ramon) [inline src/index.tsx]: you mean, curly braces; good call, addressed on: https://github.com/bespoken/dashboard/pull/355/commits/e4099d42a373f58654daee8ee05e0b682cc1658e

**Pull request merged**